frontend/src/component/common/ConstraintAccordion/ConstraintAccordionEdit/ConstraintAccordionEditBody/RestrictiveLegalValues/RestrictiveLegalValues.tsx

```diff
@@ -35,35 +36,59 @@ const createValuesMap = (values: string[]): IValuesMap => {
     }, {});
 };
 
+export const getLegalValueSet = (values: ILegalValue[]) => {
+    return new Set(values.map(({ value }) => value));
+};
+
+export const getIllegalValues = (
+    constraintValues: string[],
+    deletedLegalValues: ILegalValue[]
+) => {
+    const deletedValuesSet = getLegalValueSet(deletedLegalValues);
+
+    return constraintValues.filter(value => deletedValuesSet.has(value));
+};
+
 export const RestrictiveLegalValues = ({
     data,
     values,
     setValues,
     error,
     setError,
+    constraintValues,
 }: IRestrictiveLegalValuesProps) => {
     const [filter, setFilter] = useState('');
     const { legalValues, deletedLegalValues } = data;
+
     const filteredValues = filterLegalValues(legalValues, filter);
 
     // Lazily initialise the values because there might be a lot of them.
     const [valuesMap, setValuesMap] = useState(() => createValuesMap(values));
     const { classes: styles } = useThemeStyles();
 
-    useEffect(() => {
-        setValuesMap(createValuesMap(values));
-    }, [values, setValuesMap]);
-
     const cleanDeletedLegalValues = (constraintValues: string[]): string[] => {
-        const deletedValuesSet = new Set(
-            deletedLegalValues.map(({ value }) => value)
-        );
+        const deletedValuesSet = getLegalValueSet(deletedLegalValues);
         return (
             constraintValues?.filter(value => !deletedValuesSet.has(value)) ||
             []
         );
     };
 
+    const illegalValues = getIllegalValues(
+        constraintValues,
+        deletedLegalValues
+    );
+
+    useEffect(() => {
+        setValuesMap(createValuesMap(values));
+    }, [values, setValuesMap]);
```

Comment:
should createValuesMap also be here?

Reply:
This was just moved around, was here from before. Now that you pointed it out, I will take a look.